edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Docker/DockerConfig.cs

```diff
@@ -66,7 +64,39 @@ public bool Equals(DockerConfig other)
                    CompareCreateOptions(this.CreateOptions, other.CreateOptions);
         }
 
-        static bool CompareCreateOptions(CreateContainerParameters a, CreateContainerParameters b)
+        internal static CreateContainerParameters GetCreateOptions(string createOptions)
+        {
+            CreateContainerParameters createContainerParameters = null;
+            if (!string.IsNullOrWhiteSpace(createOptions) && !createOptions.Equals("null", StringComparison.OrdinalIgnoreCase))
+            {
+                createContainerParameters = JsonConvert.DeserializeObject<CreateContainerParameters>(createOptions);
+            }
+
+            return createContainerParameters ?? new CreateContainerParameters();
+        }
+
+        internal static string ValidateAndGetImage(string image)
```

Comment:
Is this fixing the issue with appending `latest` tag if no tag is present?

Reply:
Yes, that is correct.